server/src/main/java/org/opensearch/gateway/GatewayMetaState.java

```diff
@@ -774,19 +786,27 @@ public void markLastAcceptedStateAsCommitted() {
                 assert lastAcceptedState != null : "Last accepted state is not present";
                 assert lastAcceptedManifest != null : "Last accepted manifest is not present";
                 ClusterState clusterState = lastAcceptedState;
+                Metadata.Builder metadataBuilder = commitVotingConfiguration(lastAcceptedState);
                 if (lastAcceptedState.metadata().clusterUUID().equals(Metadata.UNKNOWN_CLUSTER_UUID) == false
                     && lastAcceptedState.metadata().clusterUUIDCommitted() == false) {
-                    Metadata.Builder metadataBuilder = Metadata.builder(lastAcceptedState.metadata());
+                    if (metadataBuilder == null) {
+                        metadataBuilder = Metadata.builder(lastAcceptedState.metadata());
+                    }
                     metadataBuilder.clusterUUIDCommitted(true);
                     clusterState = ClusterState.builder(lastAcceptedState).metadata(metadataBuilder).build();
                 }
-                final RemoteClusterStateManifestInfo committedManifestDetails = remoteClusterStateService.markLastStateAsCommitted(
-                    clusterState,
-                    lastAcceptedManifest
-                );
-                lastAcceptedManifest = committedManifestDetails.getClusterMetadataManifest();
+                if (clusterState.getNodes().isLocalNodeElectedClusterManager()) {
+                    final RemoteClusterStateManifestInfo committedManifestDetails = remoteClusterStateService.markLastStateAsCommitted(
```

Comment:
If coordination metadata is changing because of `commitVotingConfiguration`, then we should upload this voting configuration as well, right ?
The current method `remoteClusterStateService.markLastStateAsCommitted` only uploads the manifest.

Reply:
Yeah, I guess we need to update the flow in RemoteClusterStateService as well, where if we are changing the voting config, we upload that to remote as well. Which is not being done write now.

Reply:
Updated to upload coordination metadata in commit phase if voting config is updated